addons/social_share/models/campaign.py

```diff
@@ -0,0 +1,230 @@
+                'source_id': campaign.source_id.id,
+            } for campaign in campaign_ids
+        ])
+
+        mail_template_ids = self.env['mail.template'].create([{
+            'name': f'snshare {campaign.name} template',
+            'model_id': campaign.model_id.id if campaign.model_id else False,
+            'body_html': Markup(
+                '<div id="message"></div>'
+                f"""<a t-att-href="object.env['snshare.campaign'].browse({campaign.id})._get_url(object.id)" class="o_no_link_popover">{_("Your Card")}</a>"""
+            ),
+        } for campaign in campaign_ids])
+
+        for campaign, tracker, template in zip(campaign_ids, link_tracker_ids, mail_template_ids):
+            campaign.write({'link_tracker_id': tracker.id, 'mail_template_id': template.id})
+        return campaign_ids
+
+    def unlink(self):
+        self.mail_template_id.unlink()
+        return super().unlink()
+
+    def write(self, vals):
+        write_ret = super().write(vals)
+        for campaign in self:
+            link_tracker_vals = {}
+            if 'source_id' in vals:
+                link_tracker_vals['source_id'] = campaign.source_id
+            if 'target_url' in vals:
+                link_tracker_vals['url'] = campaign.target_url
+            if 'model_id' in vals:
+                campaign.mail_template_id.model_id = campaign.model_id
+            if link_tracker_vals.keys():
+                campaign.link_tracker_id.write(link_tracker_vals)
+        return write_ret
+
+    @api.depends('link_tracker_id.count')
+    def _compute_target_url_click_count(self):
+        for campaign in self:
+            campaign.target_url_click_count = self.link_tracker_id.count
+
+    @api.depends('share_url_ids.visited')
+    def _compute_share_url_click_count(self):
+        for campaign in self:
+            campaign.share_url_click_count = len(campaign.share_url_ids.filtered('visited'))
+
+    @api.depends('share_url_ids.shared')
+    def _compute_share_url_share_count(self):
+        for campaign in self:
+            campaign.share_url_share_count = len(campaign.share_url_ids.filtered('shared'))
+
+    @api.depends('link_tracker_id.short_url')
+    def _compute_target_url_redirected(self):
+        for campaign in self:
+            campaign.target_url_redirected = campaign.link_tracker_id.short_url or campaign.target_url
+
+    @api.depends('model')
+    def _compute_model_id(self):
+        model_from_name = self.env['ir.model'].search([('model', 'in', self.mapped('model'))]).grouped('model')
+        for campaign in self:
+            campaign.model_id = model_from_name.get(campaign.model, False)
+
+    @api.depends('template_id.body', 'element_ids')
+    def _compute_image(self):
+        campaigns_with_template = self.filtered('template_id.body')
+        for campaign, image in zip(campaigns_with_template, campaigns_with_template._get_images_b64(with_demo_values=True)):
```

Comment:
Ah yes, with_demo_values is close to what I had in mind for the preview of all sections in the preview. Could be worth a comment on __get_bodies_ docstring. demo_values could be updated at testing if LNA wants to show more stuff.